src/main/java/com/synopsys/integration/detect/tool/detector/DetectorToolResult.java

```diff
@@ -37,13 +37,57 @@
 import com.synopsys.integration.util.NameVersion;
 
 public class DetectorToolResult {
-    public Optional<NameVersion> bomToolProjectNameVersion = Optional.empty();
-    public List<DetectCodeLocation> bomToolCodeLocations = new ArrayList<>();
+    private final Optional<NameVersion> bomToolProjectNameVersion;
+    private final List<DetectCodeLocation> bomToolCodeLocations;
 
-    public Set<DetectorType> applicableDetectorTypes = new HashSet<>();
-    public Set<DetectorType> failedDetectorTypes = new HashSet<>();
+    private final Set<DetectorType> applicableDetectorTypes;
+    private final Set<DetectorType> failedDetectorTypes;
 
-    public Optional<DetectorEvaluationTree> rootDetectorEvaluationTree = Optional.empty();
-    public Map<CodeLocation, DetectCodeLocation> codeLocationMap = new HashMap<>();
+    private final Optional<DetectorEvaluationTree> rootDetectorEvaluationTree;
+    private final Map<CodeLocation, DetectCodeLocation> codeLocationMap;
+
+    public DetectorToolResult(final NameVersion bomToolProjectNameVersion, final List<DetectCodeLocation> bomToolCodeLocations, final Set<DetectorType> applicableDetectorTypes,
+        final Set<DetectorType> failedDetectorTypes, final DetectorEvaluationTree rootDetectorEvaluationTree,
+        final Map<CodeLocation, DetectCodeLocation> codeLocationMap) {
+        this.bomToolProjectNameVersion = Optional.of(bomToolProjectNameVersion);
+        this.bomToolCodeLocations = bomToolCodeLocations;
+        this.applicableDetectorTypes = applicableDetectorTypes;
+        this.failedDetectorTypes = failedDetectorTypes;
+        this.rootDetectorEvaluationTree = Optional.of(rootDetectorEvaluationTree);
+        this.codeLocationMap = codeLocationMap;
+    }
+
+    public DetectorToolResult() {
```

Comment:
Who creates this tool result?

Reply:
DetectorTool uses parameter-less constructor on line 95